cfy_cluster_setup/utils.py

```diff
@@ -191,3 +203,64 @@ def yum_is_present():
 
 def current_host_ip():
     return socket.gethostbyname(socket.getfqdn())
+
+
+def _check_ssl_file(filename, kind='Key'):
+    """Does the cert/key file valid?"""
+    if kind == 'Key':
+        check_command = ['openssl', 'rsa', '-in', filename, '-check', '-noout']
+    elif kind == 'Cert':
+        check_command = ['openssl', 'x509', '-in', filename, '-noout']
+    else:
+        raise ValueError('Unknown kind: {0}'.format(kind))
+    proc = run(check_command, ignore_failures=True)
+    if proc.returncode != 0:
+        raise ValidationError('{0} file {1} is invalid'.format(kind, filename))
+
+
+def check_cert_key_match(cert_filename, key_filename):
+    """Check the cert_filename matches the key_filename"""
+    _check_ssl_file(key_filename, kind='Key')
+    _check_ssl_file(cert_filename, kind='Cert')
+    key_modulus_command = [
```

Comment:
I notice all the openssl commands you're using so far are using a -noout and -in; and are either x509 or rsa. Maybe there should be a function for running openssl commands that auto-injects those parts?

Reply:
Correct, created an `openssl_command` function.